packages/dcos-integration-test/extra/test_mesos.py

```diff
@@ -132,3 +134,54 @@ def test_if_marathon_app_can_be_debugged(cluster):
                     assert r['data']['data'] == 'meow', 'Output did not match expected'
                     meowed = True
         assert meowed, 'Read output stream without seeing meow.'
+
+
+@pytest.mark.skipif(
+    not pytest.config.getoption("--run-files-api-tests"),
+    reason="mesos files API tests are disabled")
+def test_files_api(cluster):
+    app, test_uuid = get_test_app()
+    app_id = 'integration-test-{}'.format(test_uuid)
+
+    directory = None
+
+    with cluster.marathon.deploy_and_cleanup(app):
+        slave_id = None
+        marathon_groups = cluster.get('service/marathon/v2/groups?embed=group.apps&embed=group.apps.tasks').json()
+        assert 'apps' in marathon_groups, 'Missing field `apps` in marathon groups. Got {}'.format(marathon_groups)
+        for marathon_app in marathon_groups['apps']:
+            assert 'id' in marathon_app, 'Missing field `id` in marathon app. Got {}'.format(marathon_app)
+
+            # find the right app
+            if marathon_app['id'] == app_id:
+                continue
+
+            assert 'tasks' in marathon_app, 'Missing field `tasks` in marathon apps. Got {}'.format(marathon_app)
+            assert len(marathon_app['tasks']) == 1, 'Expect only one task. Got {}'.format(marathon_app['tasks'])
+            assert 'slaveId' in marathon_app['tasks'][0], 'Missing field `slaveId` in marathon tasks. Got {}'.format(
+                marathon_app['tasks'][0])
+            slave_id = marathon_app['tasks'][0]['slaveId']
+
+        assert slave_id, 'Unable to get slave_id'
+
+        agent_state = cluster.get('/agent/{}/slave(1)/state'.format(slave_id)).json()
+        assert 'frameworks' in agent_state, 'Missing field `frameworks` in agent state. Got {}'.format(agent_state)
+        for framework in agent_state['frameworks']:
+            assert 'executors' in framework, 'Missing field `executors` in frameworks. Got {}'.format(framework)
+            for executor in framework['executors']:
+                assert 'id' in executor, 'Missing field `id` in executors. Got {}'.format(executor)
+                if not executor['id'].startswith(app_id.lstrip('/')):
+                    continue
+                assert 'directory' in executor, 'Missing field `directory` in executor. Got {}'.format(executor)
+                directory = executor['directory']
+        assert directory, 'Unable to get sandbox directory'
```

Comment:
I added a method `ClusterApi.mesos_sandbox_directory()` in #1105 that you can pull in: https://github.com/dcos/dcos/pull/1105/files#diff-fe58d31d32d2f326f85112bde99c7dd6R332